scripts/prepare_disk.sh

```diff
@@ -22,6 +22,58 @@ usage()
     echo "NOTE: This script is destructive and will destroy your data."
 }
 
+is_integer()
+{
+    test "${1}" -eq "${1}" 2> /dev/null
+}
+
+is_comment()
+{
+    test -z "${1%%#*}"
+}
+
+# A bit counter intuitive, a function can either return an exit code,
+# or set a global variable, or echo something, in this case the
+# exit code is 0 if a resize is required.
```

Comment:
This I guess is over-explaining things a little, in that '0 is ok, anything else is error' and thus when reading `if is_resize_needed` the required exit value is counter intuitive as 1, rather then 0 to prevent having to read `if ! is_resize_needed`

Alternatively we could have used `no_resize_needed`, which would have resulted in the more expected exit value of 0 to indicate success of a function, lessening the (imo)slightly confusing comment.